packages/manager/modules/manager-pci-common/src/components/shape-input/ShapeInput.component.tsx

```diff
@@ -355,7 +358,13 @@ export const ShapesInputComponent = function ShapesInputComponent<T>({
                   isMobile={isMobile}
                 />
               }
-              isOpen={!groupName}
+              isOpen={isAccordionOpen[groupName || 'none']}
+              onToggle={() => {
+                const name = groupName || 'none';
+                setIsAccordionOpen({
+                  [name]: !isAccordionOpen[name],
+                });
+              }}
```

Comment:
Could be isolated into a `useCallbak` to prevent from re-computing

Reply:
since "name" comes from a map in the template i'm not sure it wouid be easiy doable, since it still need to be passed to the function dynamically, do you see another way ?

Reply:
We can use currification to achieve that, but I see that it does not work with `useCallback`  as we can lose reference of the curried function at re-render :/
Maybe using `useMemo`, but to be tested :)

```ts
  const handleToggle = useMemo(
    () => (groupName: string) => () => {
      setIsAccordionOpen((prev) => ({
        [groupName || 'none']: !prev[groupName || 'none'],
      }));
    },
    [setIsAccordionOpen],
  );
```

And it's simply usage in jsx: 

```ts
 onToggle={handleToggle(groupName)}
```

Reply:
just added

```typescript
type TGroupName = 'none' | string;
```

please let me know if it's fine for you